### DISCUSSION TITLE
Gas cost per transaction in ETH main net

### DISCUSSION THREAD

**Discussion by leonardolopezcallejo** [Cyfrin/foundry-full-course-cu#1804] [General]
Hi, just went through the point in this amazing course where Patrick shows some techniques about saving gas by optimizing code (https://youtu.be/umepbfKp5rI?t=21526), cool, but my question is, are those numbers real? can a transaction cost around 9 USD ? I asked ChatGPT and said that, as average, it costs 0.00105 ETH, that means 4$ as today, which looks to me also pretty high

are my calculations and assumptions correct? it's really disappointing me and making me rethink my solution

**Comment by EngrPips** [comment]
Hello @zendeldawn, I don't think your calculations are wrong at all. Sometimes when I try to make a single swap on a decentralized exchange, it costs $15 or above. The more congested the blockchain is the more costly transactions become.

**Comment by urosognjenovic** [comment]
During the 2021 and 2022 bull run and NFT craze, transaction fees were as high as a couple hundred dollars during peak times. That's why there are many layer 2 scaling solutions that should reduce the transaction fees and increase the transaction throughput.

You can view the historical average transaction fee values on this website: https://bitinfocharts.com/comparison/ethereum-transactionfees.html#alltime

**Comment by simrangill1551** [comment]
It will be 100's of dollars soon , this is cheap compared to 2021 prices

**Reply by EngrPips** [reply]
Lol, High Gas fee is not suitable for my health.
